coriolisclient/tests/cli/test_replicas.py

```diff
@@ -0,0 +1,586 @@
+# Copyright 2024 Cloudbase Solutions Srl
+# All Rights Reserved.
+
+import os
+from unittest import mock
+
+from cliff import command
+from cliff import lister
+from cliff import show
+
+from coriolisclient.cli import replica_executions
+from coriolisclient.cli import replicas
+from coriolisclient.cli import utils as cli_utils
+from coriolisclient.tests import test_base
+
+
+class ReplicaFormatterTestCase(test_base.CoriolisBaseTestCase):
+    """Test suite for the Coriolis Client Replica Formatter."""
+
+    def setUp(self):
+        super(ReplicaFormatterTestCase, self).setUp()
+        self.replica = replicas.ReplicaFormatter()
+
+    def test_get_sorted_list(self):
+        obj1 = mock.Mock()
+        obj2 = mock.Mock()
+        obj3 = mock.Mock()
+        obj1.created_at = "date1"
+        obj2.created_at = "date2"
+        obj3.created_at = "date3"
+        obj_list = [obj2, obj1, obj3]
+
+        result = self.replica._get_sorted_list(obj_list)
+
+        self.assertEqual(
+            [obj1, obj2, obj3],
+            result
+        )
+
+    def test_format_last_execution(self):
+        obj = mock.Mock()
+        obj.executions = None
+
+        result = self.replica._format_last_execution(obj)
+
+        self.assertEqual(
+            "",
+            result
+        )
+
+        execution1 = mock.Mock()
+        execution2 = mock.Mock()
+        execution3 = mock.Mock()
+        execution1.created_at = "date1"
+        execution2.created_at = "date2"
+        execution3.created_at = "date3"
+        execution1.to_dict.return_value = {
+            "id": "mock_id1",
+            "status": "mock_status1"
+        }
+        execution2.to_dict.return_value = {
+            "id": "mock_id2",
+            "status": "mock_status2"
+        }
+        execution3.to_dict.return_value = {
+            "id": "mock_id3",
+            "status": "mock_status3"
+        }
+        obj.executions = [execution1, execution3, execution2]
+
+        result = self.replica._format_last_execution(obj)
+
+        self.assertEqual(
+            "mock_id3 mock_status3",
+            result
+        )
+
+    def test_get_formatted_data(self):
+        obj = mock.Mock()
+        obj.id = mock.sentinel.id
+        obj.last_execution_status = mock.sentinel.last_execution_status
+        obj.instances = ["mock_instance3", "mock_instance1", "mock_instance2"]
+        obj.notes = mock.sentinel.notes
+        obj.created_at = mock.sentinel.created_at
+
+        result = self.replica._get_formatted_data(obj)
+
+        self.assertEqual(
+            (
+                mock.sentinel.id,
+                ('mock_instance3%(ls)smock_instance1%(ls)smock_instance2'
+                 % {"ls": os.linesep}),
```

Comment:
This could fail if run on Windows. Please use `'\n'` instead
```suggestion
                 % {"ls": "\n"}),
```